src/lucky/routeable.cr

```diff
@@ -175,6 +175,19 @@ module Lucky::Routeable
       route(*args, **named_args).url
     end
 
+    def self.url_without_query_params(
+    {% for param in path_params %}
+      {{ param.gsub(/:/, "").id }},
+    {% end %}
+    )
+      path = path_from_parts(
+        {% for param in path_params %}
+          {{ param.gsub(/:/, "").id }},
+        {% end %}
+      )
```

Comment:
I wish I knew of a good way to DRY up macros like this, but I don't know of a way :(